File: Data Structures & Algorithms/top-k-elements-in-list/submission-4.py
class Solution:
    def topKFrequent(self, nums: List[int], k: int) -> List[int]:
        freq_dict = {}

        for n in nums:
            freq_dict[n] = 1 + freq_dict.get(n, 0)

        buckets = [[] for n in range(len(nums)+1)]

        for num, freq in freq_dict.items():
            buckets[freq].append(num)

        res = []
        for i in range(len(buckets)-1, 0, -1):
            for num in buckets[i]:
                res.append(num)
                if len(res) == k:
                    return res

File: Data Structures & Algorithms/binary-search/submission-4.cpp
class Solution {
public:
    int search(vector<int>& nums, int target) {
        int l=0;
        int r=nums.size()-1;
        int middle;
        while (l<=r) {
            middle = (l+r)/2;
            if (nums[middle] == target) return middle;
            else if (nums[middle] > target) r = middle-1;
            else l = middle+1;
        }
        return -1;
    }
};
